modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/StyleBookModal.js

```diff
@@ -0,0 +1,217 @@
+	value: '-1',
+};
+
+export default function StyleBookModal({
+	addStyleBookEntryURL,
+	frontendTokenDefinitions = [],
+	namespace,
+	onModalClose,
+}) {
+	const {observer, onClose} = useModal({onClose: () => onModalClose()});
+
+	const [loading, setLoading] = useState(false);
+
+	const [errors, setErrors] = useReducer(
+		(value, nextValue) => ({...value, ...nextValue}),
+		{frontendTokenDefinition: null, name: null}
+	);
+
+	const [name, setName] = useState('');
+	const [frontendTokenDefinition, setFrontendTokenDefinition] =
+		useState(null);
+
+	const formRef = useRef(null);
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+
+		const errors = validateFields(name, frontendTokenDefinition);
+
+		if (Object.keys(errors).length) {
+			setErrors(errors);
+
+			return;
+		}
+
+		setLoading(true);
+
+		const body = Liferay.Util.ns(namespace, {
+			name,
+			themeId: frontendTokenDefinition.themeId,
+		});
+
+		fetch(addStyleBookEntryURL, {
+			body: objectToFormData(body),
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((responseContent) => {
+				if (responseContent.error) {
+					setLoading(false);
+					setErrors(responseContent.error);
+				}
+				else if (responseContent.redirectURL) {
+					navigate(responseContent.redirectURL, {
+						beforeScreenFlip: onClose,
+					});
+				}
+			})
+			.catch(() =>
+				setErrors({
+					other: Liferay.Language.get(
+						'an-unexpected-error-occurred-while-creating-the-style-book'
+					),
+				})
+			);
+	};
+
+	const nameId = `${namespace}name`;
+	const frontendTokenDefinitionId = `${namespace}tokenDefinition`;
+
+	return (
+		<ClayModal observer={observer} size="md">
+			<ClayModal.Header>
+				{Liferay.Language.get('add-style-book')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				{errors.other && (
+					<ClayAlert
+						displayType="danger"
+						onClose={() => {}}
+						title={Liferay.Language.get('error')}
+					>
+						{errors.other}
+					</ClayAlert>
+				)}
+
+				<FieldBase
+					errorMessage={errors.frontendTokenDefinition}
+					id={frontendTokenDefinitionId}
+					label={Liferay.Language.get('create-style-book-for')}
+					name="frontendTokenDefinition"
+					required={true}
+				>
+					<ClaySelect
+						id={frontendTokenDefinitionId}
+						onChange={(event) => {
+							const value = event.target.value;
+
+							const tokenDefinition =
+								value === -1
+									? null
+									: frontendTokenDefinitions[value];
+
+							setFrontendTokenDefinition(tokenDefinition);
+						}}
+					>
+						<ClaySelect.Option
+							label={DEFAULT_OPTION.label}
+							value={DEFAULT_OPTION.value}
+						/>
+
+						{frontendTokenDefinitions.map(
+							(tokenDefinition, index) => (
+								<ClaySelect.Option
+									key={tokenDefinition.themeId}
+									label={tokenDefinition.name}
+									value={index}
+								/>
+							)
+						)}
+					</ClaySelect>
+
+					<p className="text-secondary">
```

Comment:
Fieldbase has the prop `helpMessage` where you can pass this message in.

Reply:
I used errorMessage for this instead, this one appears red with an exclamation mark red circle icon. I tried switching to helpMessage but it is just gray text. Is it preferred to use helpMessage instead of errorMessage?

Reply:
You need both. `helpMessage` is the gray text and `errorMessage` shows when something is wrong.

We also need to clear the error message on `<ClaySelect onChange={}` 
![fetd](https://github.com/user-attachments/assets/b4703813-8bfe-4eb7-8287-c48d7bab6e64)

Reply:
Ah sorry thank you for explaining, I misunderstood. I commented the fix for this on the new ticket as I am not sure if I can just push it myself